src/metal_guardian.c

```diff
@@ -0,0 +1,476 @@
+#include "metal_guardian.h"
+
+#include "functions.h"
+#include "kirby.h"
+#include "object.h"
+#include "task.h"
+
+struct MetalGuardianLaser;
+
+typedef u32 (*MetalGuardianFunc)(struct MetalGuardianLaser*);
+
+struct MetalGuardianLaser {
+    struct ObjectBase base;
+    MetalGuardianFunc unk78;
+    MetalGuardianFunc unk7C;
+    MetalGuardianFunc unk80;
+    u32 unk84;
+    u8 unk88;
+    u8 unk89;
+    u16 unk8A;
+};
+
+static const struct Unk_08353510 gUnk_08354A74[] = {
+    { -0x40, 0x100, 0x0, 0x0, 0x10, 0x2 },
+    { -0xa0, 0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { -0x100, 0x40, 0x0, 0x0, 0x10, 0x2 },
+    { -0x100, -0x40, 0x0, 0x0, 0x10, 0x2 },
+    { -0xa0, -0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { -0x40, -0x100, 0x0, 0x0, 0x10, 0x2 },
+    { 0x40, -0x100, 0x0, 0x0, 0x10, 0x2 },
+    { 0xa0, -0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { 0x100, -0x40, 0x0, 0x0, 0x10, 0x2 },
+    { 0x100, 0x40, 0x0, 0x0, 0x10, 0x2 },
+    { 0xa0, 0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { 0x40, 0x100, 0x0, 0x0, 0x10, 0x2 },
+    { 0x40, 0x100, 0x0, 0x0, 0x10, 0x2 },
+    { 0xa0, 0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { 0x100, 0x40, 0x0, 0x0, 0x10, 0x2 },
+    { 0x100, -0x40, 0x0, 0x0, 0x10, 0x2 },
+    { 0xa0, -0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { 0x40, -0x100, 0x0, 0x0, 0x10, 0x2 },
+    { -0x40, -0x100, 0x0, 0x0, 0x10, 0x2 },
+    { -0xa0, -0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { -0x100, -0x40, 0x0, 0x0, 0x10, 0x2 },
+    { -0x100, 0x40, 0x0, 0x0, 0x10, 0x2 },
+    { -0xa0, 0xa0, 0x0, 0x0, 0x10, 0x2 },
+    { -0x40, 0x100, 0x0, 0x0, 0x10, 0x2 },
+    { 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 },
+};
+
+static const u8 gUnk_08354BA0[8] = { 0xf8, 0xf7, 0xfa, 0xfc, 0x2, 0xfe, 0x2, 0xfe };
```

Comment:
Should be `s8` based on the access. And the elements need to be changed respectively.